atomicapp/requirements.py

```diff
@@ -61,12 +61,12 @@ def _exec(self, action):
             requirement_function = self._find_requirement_function_name(key_name)
 
             # Check to see if the function exists in the provider,
-            # if it does not: fail
+            # if it does not: warn the user
             try:
                 requirement = getattr(self.provider, requirement_function)
             except AttributeError:
-                raise RequirementFailedException(
-                    "Requirement %s does not exist within %s." %
+                logger.warning(
+                    "Requirement %s does not exist within %s. Skipping." %
                     (requirement_function, self.provider))
```

Comment:
you need to return here. otherwise you get this:

```
UnboundLocalError: local variable 'requirement' referenced before assignment
```

because of the next statement that executes.